contracts/Staking.sol

```diff
@@ -0,0 +1,488 @@
+pragma solidity 0.6.4;
+
+import "./System.sol";
+import "./lib/Memory.sol";
+import "./lib/BytesToTypes.sol";
+import "./interface/IApplication.sol";
+import "./interface/ICrossChain.sol";
+import "./interface/IParamSubscriber.sol";
+import "./interface/ITokenHub.sol";
+import "./lib/CmnPkg.sol";
+import "./lib/SafeMath.sol";
+import "./lib/RLPEncode.sol";
+import "./lib/RLPDecode.sol";
+import "./interface/IStaking.sol";
+
+contract Staking is IStaking, System, IParamSubscriber, IApplication {
+  using SafeMath for uint256;
+  using RLPEncode for *;
+  using RLPDecode for *;
+
+  struct DelegateAckPackage {
+    address delegator;
+    address validator;
+    uint256 amount;
+    uint8 errCode;
+  }
+
+  struct UndelegateAckPackage {
+    address delegator;
+    address validator;
+    uint256 amount;
+    uint8 errCode;
+  }
+
+  struct RedelegateAckPackage {
+    address delegator;
+    address valSrc;
+    address valDst;
+    uint256 amount;
+    uint8 errCode;
+  }
+
+  struct TransferInRewardSynPackage {
```

Comment:
I think it is not proper to use wording like `TransferIn` or `TransferOut`.

Reply:
Fixed